tests/UI/campaigns/productV2/functional/01_filterProducts.ts

```diff
@@ -0,0 +1,295 @@
+
+        const textMessage = await productsPage.getAlertDangerBlockContent(page);
+        await expect(textMessage).to.equal(test.args.alertDanger);
+      });
+    });
+
+    [
+      {
+        args: {
+          identifier: 'filterIDMinMax',
+          filterBy: 'id_product',
+          filterValue: {min: 5, max: 10},
+          filterType: 'input',
+        },
+      },
+      {
+        args: {
+          identifier: 'filterName',
+          filterBy: 'product_name',
+          filterValue: Products.demo_14.name,
+          filterType: 'input',
+        },
+      },
+      {
+        args: {
+          identifier: 'filterReference',
+          filterBy: 'reference',
+          filterValue: Products.demo_1.reference,
+          filterType: 'input',
+        },
+      },
+      {
+        args: {
+          identifier: 'filterCategory',
+          filterBy: 'category',
+          filterValue: Categories.women.name,
+          filterType: 'input',
+        },
+      },
+      {
+        args: {
+          identifier: 'filterPriceMinMax',
+          filterBy: 'price',
+          filterValue: {min: 5, max: 10},
+          filterType: 'input',
+        },
+      },
+      {
+        args: {
+          identifier: 'filterQuantityMinMax',
+          filterBy: 'quantity',
+          filterValue: {min: 100, max: 1000},
+          filterType: 'input',
+        },
+      },
+      {
+        args: {
+          identifier: 'filterStatusYes',
+          filterBy: 'active',
+          filterValue: 'Yes',
+          filterType: 'select',
+        },
+      },
+    ].forEach((test) => {
+      it(`should filter list by '${test.args.filterBy}' and check result`, async function () {
+        await testContext.addContextItem(this, 'testIdentifier', `${test.args.identifier}`, baseContext);
+
+        await productsPage.filterProducts(page, test.args.filterBy, test.args.filterValue, test.args.filterType);
+
+        const numberOfProductsAfterFilter = await productsPage.getNumberOfProductsFromList(page);
+
+        if (test.args.filterBy === 'active') {
+          await expect(numberOfProductsAfterFilter).to.be.above(0);
+        } else {
+          await expect(numberOfProductsAfterFilter).to.be.below(numberOfProducts);
+        }
+
+        for (let i = 1; i <= numberOfProductsAfterFilter; i++) {
+          const textColumn = await productsPage.getTextColumn(page, test.args.filterBy, i);
+
+          if (test.args.filterBy === 'id_product' || test.args.filterBy === 'price'
+            || test.args.filterBy === 'quantity') {
+            await expect(textColumn).to.within(test.args.filterValue.min, test.args.filterValue.max);
+          } else if (test.args.filterBy === 'active' && test.args.filterValue === 'Yes') {
+            await expect(textColumn).to.be.true;
+          } else {
+            await expect(textColumn).to.be.contain(test.args.filterValue);
+          }
```

Comment:
Why not create a `test.args.filterResult` => 'boolean', 'text' or 'number' which use each check rather using the the `filterBy` key ?

Reply:
Thank you @Progi1984 :)
Yes I thought about the new arg but the problem  in this case is  : we have different types of comparison (`.to.within`, `to.be.true`,` .to.be.contain` )
Do you have a solution for that ?

Reply:
Use  a switch case for each type and a default triggers an error.